src/rosie.js

```diff
@@ -399,6 +399,17 @@ Factory.util = (function() {
         }
       }
       return dest;
+    },
+
+    /**
+     * Deep clones given object.
+     *
+     * @private
+     * @param {object} object
+     * @return {object}
+     */
+    deepClone: function(object) {
+      return JSON.parse(JSON.stringify(object));
```

Comment:
Yes but I'm not sure we would want to bring in lodash just for this. Unless there's no other way around the cases @eventualbuddha mentioned